src/Microsoft.AspNetCore.OData/Extensions/HttpRequestExtensions.cs

```diff
@@ -388,26 +388,16 @@ public static void DeleteRequestContainer(this HttpRequest request, bool dispose
         /// <returns></returns>
         private static IServiceScope CreateRequestScope(this HttpRequest request, string routeName)
         {
-            IServiceProvider rootContainer;
-            if (string.IsNullOrEmpty(routeName))
+            IPerRouteContainer perRouteContainer = request.HttpContext.RequestServices.GetRequiredService<IPerRouteContainer>();
+            if (perRouteContainer == null)
             {
-                // For HTTP routes, use the default request services.
-                rootContainer = request.HttpContext.RequestServices;
+                throw Error.ArgumentNull("routeName");
```

Comment:
We're throwing argument null exception for "routeName" when perRouteContainer is null? Should this one be an exception for perRouteContainer and a separate if-statement check for routeName above?